rest/adminapitest/admin_api_test.go

```diff
@@ -241,6 +241,39 @@ func TestLoggingKeys(t *testing.T) {
 	assert.Equal(t, map[string]interface{}{}, noLogKeys)
 }
 
+func TestServerlessChangesEndpointLimit(t *testing.T) {
+	base.RequireNumTestBuckets(t, 2)
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyReplicate, base.KeyHTTP, base.KeyHTTPResp, base.KeySync, base.KeySyncMsg)
+	rt := rest.NewRestTester(t, nil)
+	defer rt.Close()
+
+	resp := rt.SendAdminRequest(http.MethodPut, "/_config", `{"max_concurrent_connections" : 2}`)
+	rest.RequireStatus(t, resp, http.StatusOK)
+
+	for i := 0; i < 200; i++ {
+		_ = rt.PutDoc(fmt.Sprint(i), `{"source":"rt","channels":["alice"]}`)
+	}
+
+	// send some changes requests in go routines to run concurrently along with test
+	go func() {
+		resp1 := rt.SendAdminRequest(http.MethodGet, "/{{.keyspace}}/_changes?feed=longpoll&since=999999&timeout=100000", "")
+		rest.RequireStatus(t, resp1, http.StatusOK)
+	}()
+
+	go func() {
+		resp2 := rt.SendAdminRequest(http.MethodGet, "/{{.keyspace}}/_changes?feed=longpoll&since=999999&timeout=100000", "")
+		rest.RequireStatus(t, resp2, http.StatusOK)
+	}()
```

Comment:
I'm always cautious about creating long running go routines in a test, if you drop `rt.WaitForActiveReplicatorCount` - these can run the background.

I'm not sure how far I want to take this, but I'd be inclined to put these in a waitgroup so we know the assertions are actually run, and make sure they are finished at the end of the test. The downside of this is that the test will time out if it isn't asserting correctly. OTOH, these assertions may never get hit because you are asking for them to wait 1000000.

I think that tearing down a database in a defer will destroy these goroutines, but you should make sure in the test.